### DISCUSSION TITLE
How to align this like vim-easy-align?

### DISCUSSION THREAD

**Discussion by 9mm** [nvim-mini/mini.nvim#457] [Q&A] [answered]
I used to use this plugin and it worked quite well. For example this block in ruby:

```
      t.references :lead, null: false, foreign_key: true, type: :uuid
      t.integer    :type, null: false
      t.jsonb :response, null: false, default: {}
```

I want it to align with all the `:` in alignment, and all the `,` in alignment.

In old plugin, I would do 2 passes, the first one: `ga `

```
      t.references :lead, null: false, foreign_key: true, type: :uuid
      t.integer    :type, null: false
      t.jsonb      :response,  null: false, default: {}
```

and then `ga,`

```
      t.references :lead,     null: false, foreign_key: true, type: :uuid
      t.integer    :type,     null: false
      t.jsonb      :response, null: false, default: {}
```

But in mini.nvim, i'm trying to figure out how to align things int he same way. This is what I get:

```
      t.references :lead, null: false, foreign_key: true, type: :uuid
      t.integer    :type, null: false
      t.jsonb :response,  null: false, default: {}
```

```
      t.references :lead,     null: false, foreign_key: true, type: :uuid
      t.integer    :type,     null: false
      t.jsonb      :response, null: false, default:     {}
```

I think the key of my question lies in the fact of how many "columns" its formatting. I don't see a way I can configure htis in the docs unless I'm missing something?

I believe nvim-easy-align is 1 column by default.

I don't necessarily have to recreate the old behavior, but maybe you could tell me what is the best way to use your plugin with your new approach?

**Comment by echasnovski** [comment, marked as answer]
> I think the key of my question lies in the fact of how many "columns" its formatting. I don't see a way I can configure htis in the docs unless I'm missing something?
> 
> I believe nvim-easy-align is 1 column by default.

Yes, 'mini.align' indeed aligns by all columns by default. This seems to me like a more reasonable behavior.

To control which parts get aligned, use [`filter()`](https://github.com/echasnovski/mini.nvim/blob/6b5a2dbbb80edeb0f4c1b507e0cd41844f5cb65e/doc/mini-align.txt#L809) step which is [mapped to `f` modifier by default](https://github.com/echasnovski/mini.nvim/blob/6b5a2dbbb80edeb0f4c1b507e0cd41844f5cb65e/doc/mini-align.txt#L443). So to replicate what you are doing with 'vim-easy-align':
- Type `gAip` to start interactive align; then `:` to enter split pattern; followed by `f` (should show "Enter filter expression" input dialog), `n==1` and `<CR>` (`<Enter>`). This should show the result you describe (except some whitespace) as after first step of 'vim-easy-align'. Confirm with `<CR>`.
- Type `gaip,` to align by `,`. This should the result you describe as after second step.

Here is [how you can make aligning only by first pair to be the default](https://github.com/echasnovski/mini.nvim/blob/6b5a2dbbb80edeb0f4c1b507e0cd41844f5cb65e/doc/mini-align.txt#L541-L546).

Does this answer your question?

**Reply by 9mm** [reply]
Wow, very cool. Yes, I agree 1 by default is weird, I like this approach, it will take getting used to but it's much  more flexible. Thank you!!
